meta-iot2000-example/recipes-example/iot2000setup/files/iot2000setup.py

```diff
@@ -0,0 +1,503 @@
+	global deviceIsIot2020
+	
+	if deviceIsIot2020:
+		interfaces = ([('eth0', getNetworkInterfaceConfiguration('eth0'))])
+	else:
+		interfaces = ([('eth0', getNetworkInterfaceConfiguration('eth0')) , ('eth1', getNetworkInterfaceConfiguration('eth1'))])
+		
+	if wifiEnabled:
+		interfaces.append(('wlan0', getNetworkInterfaceConfiguration('wlan0')))
+		
+	ret = EntryWindow(
+		gscreen,
+		"Configure Network Interfaces",
+		"Specify IP addresses for network interfaces, enter 'dhcp' to obtain address by DHCP.",
+		interfaces,
+		1,
+		70, 50,
+		['OK', ('Cancel', 'cancel', 'ESC')],
+		None)
+	
+	interfacesConfig = """# /etc/network/interfaces -- configuration file for ifup(8), ifdown(8)
+ 
+# The loopback interface
+auto lo
+iface lo inet loopback
+
+# Wired interfaces
+"""	
+	dhcpTemplate = """auto [interfaceName]
+iface [interfaceName] inet dhcp
+
+"""
+	staticTemplate = """auto [interfaceName]
+iface [interfaceName] inet static
+	address [ip]
+	netmask 255.255.255.0
+	
+"""
+
+	wirelessDhcpTemplate = """allow-hotplug wlan0
+auto wlan0
+iface wlan0 inet dhcp
+	wpa-conf /etc/wpa_supplicant.conf
+
+"""
+	wirelessStaticTemplate = """allow-hotplug wlan0
+auto wlan0
+iface wlan0 inet static
+	address [ip]
+	netmask 255.255.255.0
+	wpa-conf /etc/wpa_supplicant.conf
+
+"""
+	
+	i = 0
+	for interface in interfaces:
+		if (interface[0] == "wlan0"):
+			if (ret[1][i] == "dhcp"):
+				interfacesConfig = interfacesConfig + wirelessDhcpTemplate.replace("[interfaceName]", interface[0])
+			else:
+				interfacesConfig = interfacesConfig + wirelessStaticTemplate.replace("[interfaceName]", interface[0]).replace("[ip]", ret[1][i])
+		else:
+			if (ret[1][i] == "dhcp"):
+				interfacesConfig = interfacesConfig + dhcpTemplate.replace("[interfaceName]", interface[0])
+			else:
+				interfacesConfig = interfacesConfig + staticTemplate.replace("[interfaceName]", interface[0]).replace("[ip]", ret[1][i])
+		i += 1
+	
+	if (ret[0] == "ok"):
+		fileName = "/etc/network/interfaces"
+		backupFileName = "/etc/network/interfaces.bak"
+		copyfile(fileName, backupFileName)
+		interfacesFile = open(fileName, 'w')
+		interfacesFile.write(interfacesConfig)
+		interfacesFile.close()
+		
+		networkConfigurationChanged = 1
+		rv = ButtonChoiceWindow(
+				gscreen,
+				"Configure Network Interfaces",
+				"Your network interfaces have been reconfigured. A backup of the old configuration can be found at: " + backupFileName,
+				buttons=["OK"],
+				width=40)
+
+	displayStartScreen()
+	
+
+def expandFileSystem():
+	
+	subprocess.call("/etc/iot2000setup/expandfs.sh", stdout=open(os.devnull, 'wb')) 
+	task = subprocess.Popen("df -h | grep '/dev/root' | awk '{print $2}'", stdout=subprocess.PIPE, shell=True)
+	newPartitionSize = task.stdout.read().lstrip().rstrip()
+
+	rv = ButtonChoiceWindow(
+				gscreen,
+				"Expand File System",
+				"Successfully expanded file system. New partition size is " + newPartitionSize + ".",
+				buttons=["OK"],
+				width=40)
+
+	displayStartScreen()
```

Comment:
This particular feature should be converted to a run-once script for the first target boot. No need for human interaction then.